Stop jumping applet as numbers change

lukefromdc opened the pull request:
Hold the value labels to the largest size they have been unless the units have just been shown or hidden. This prevents the applet as a whole from jumping as temperatures (etc) change.

monsta commented:
I guess I'd need to switch to Fahrenheit to reproduce it? Because with Celsius the numbers are always two-digit, so there's no any jumping.

lukefromdc commented:
The fast way to test is to cycle between farenheit and celsius, which should jump at least one sensor between two and three digits.  I will see if the function to change the scale is reachable/readable from the code that sets the label width, if so I can modify this PR to reset the minimum width when the temperature scale is changed.

On a celsius scale, the jump is a very subtle jump of the whole applet. With regular fonts maybe only 1 px. 

On the first increase in space taken my a number, it must either jump or truncate so a hard maximum would not be a good idea. What I've done is to hold that larger size, so when the numbers take up less space the applet does not jump back in, then move in and out as the numbers vary.

lukefromdc commented:
The indents in the entire file are an ugly, messed up mix of tabs and spaces, enough to sow confusion in working on it. I am playing with that now, hoping to remove all the tabs, then gets the spaces right through the entire file.

info-cppsp commented:
You shouldn't. I intend to take care of that, and in all files, later.

lukefromdc commented:
OK, I'll just make a couple lines consistant with those around them

lukefromdc commented:
OK, I changed the spaces on the two lines below the lines I added at the top of the first modified function to tabs like the rest of the lines around them(including the ones I added). I kept my own changes consistant with the state of the file around them, seeking to avoid making this mess of tabs and spaces any worse.

info-cppsp commented:
It is not that crucial, I just meant, that it is a waste of your precious time. ;-)
(Also it should adhere to an agreed upon standard for all files.)

monsta commented:
I actually meant one-time switching to Fahrenheit and then testing how the applet behaves when switching from two to three digits and back. No switching back and forth between scales.

lukefromdc commented:
Yes, that's the simplest test

raveit65 commented:
merged https://github.com/mate-desktop/mate-sensors-applet/commit/458315df723f211956edcbbf96e8c84acc9da06c
Thx
